### PR TITLE
[WIP] Logstash Netflow module tutorial

### PR DESCRIPTION
**Pull request by ycombinator**
This PR adds a Kibana tutorial for the Logstash Netflow module.

This PR is a work in progress.

## Things still left TODO

* [x] Incorporate [writing style suggestions](https://github.com/elastic/dev/issues/844#issuecomment-342999058) from @dedemorton

## Current rendering of this tutorial

screen shot 2017-11-08 at 4 52 28 pm

### CODE REVIEW

File: src/core_plugins/kibana/server/tutorials/netflow/index.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    longDescription: 'The Logstash Netflow module simplifies the collection, normalization, and visualization of network flow data.' +
+      ' With a single command, the module parses network flow data, indexes the events into Elasticsearch, and installs a suite of Kibana' +
+      ' dashboards to get you exploring your data immediately. Logstash modules support Netflow Version 5 and 9.',
+    //iconPath: '', TODO
+    completionTimeMinutes: 10,
+    //previewImagePath: 'kibana-apache.png', TODO
+    params: [
+      {
+        'netflow.var.input.udp.port': {
+          type: 'number', // TODO: Make this a const as well?
```
Comment (ycombinator) [inline src/core_plugins/kibana/server/tutorials/netflow/index.js:21]: @nreese What do you think of making these consts/enums as well?

Comment (nreese) [reply src/core_plugins/kibana/server/tutorials/netflow/index.js:21]: good idea. I have started a new file for param type constants that you can use - https://github.com/nreese/kibana/blob/home_phase_two/src/core_plugins/kibana/common/tutorials/param_types.js

File: src/core_plugins/kibana/server/tutorials/netflow/index.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                textPre: 'Skip this step if Logstash is already installed. First time using Logstash? See the ' +
+                         '[Getting Started Guide]({config.elastic_docs.website_url}/guide/en/logstash/{config.elastic_docs.link_version}' +
+                         '/getting-started-with-logstash.html).',
+                commands: [
+                  'curl -L -O https://artifacts.elastic.co/downloads/logstash/logstash-{config.kibana.version}.tar.gz',
+                  'tar xzvf logstash-{config.kibana.version}.tar.gz'
+                ]
+              },
+              {
+                title: 'Setup the Netflow module',
```
Comment (dedemorton) [inline src/core_plugins/kibana/server/tutorials/netflow/index.js:44]: Should say "Set up" (two words). See http://grammarist.com/spelling/set-up-vs-setup/

File: src/core_plugins/kibana/server/tutorials/netflow/index.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                textPre: 'In the Logstash install directory, run the following command to setup the Netflow module.',
```
Comment (dedemorton) [inline src/core_plugins/kibana/server/tutorials/netflow/index.js:45]: same here. setup > set up.

File: src/core_plugins/kibana/server/tutorials/netflow/index.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                commands: [
+                  './bin/logstash --modules netflow --setup',
+                ],
+                textPost: 'The --setup option creates a `netflow-*` index pattern in Elasticsearch and imports' +
```
Comment (dedemorton) [inline src/core_plugins/kibana/server/tutorials/netflow/index.js:49]: Surround --setup in backtics as you have in other places.

File: src/core_plugins/kibana/server/tutorials/netflow/index.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+                  ' Kibana dashboards and visualizations. Running `--setup` is a one-time setup step. Omit this step' +
```
Comment (dedemorton) [inline src/core_plugins/kibana/server/tutorials/netflow/index.js:50]: We should standardize on the text that we use in the tutorials to explain --setup. I think the Filebeat modules use something slightly different. For UI text especially, we should be super concise.

Comment (dedemorton) [reply src/core_plugins/kibana/server/tutorials/netflow/index.js:50]: Note that here setup (one word) is used correctly. :-)

File: src/core_plugins/kibana/server/tutorials/netflow/index.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+                commands: [
+                  'curl -L -O https://artifacts.elastic.co/downloads/logstash/logstash-{config.kibana.version}.tar.gz',
+                  'tar xzvf logstash-{config.kibana.version}.tar.gz'
+                ]
+              },
+              {
+                title: 'Setup the Netflow module',
+                textPre: 'In the Logstash install directory, run the following command to setup the Netflow module.',
+                commands: [
+                  './bin/logstash --modules netflow --setup',
```
Comment (dedemorton) [inline src/core_plugins/kibana/server/tutorials/netflow/index.js:47]: Are there any platforms where the leading `./` actually required to run Logstash? Not sure, but I'm able to run Logstash on mac with `bin/logstash`, and that's what we use all over the Logstash docs. TBH, I wish Beats could do the same.

**Review by dedemorton — COMMENTED** [review]
@ycombinator Looks like this was merged before I had a chance to review it. I've added my comments anyhow so they get captured somewhere.

**Comment by nreese** [comment]
@dedemorton I have integrated your changes into https://github.com/elastic/kibana/pull/14749. You should do your reviews against home_phase_two PR https://github.com/elastic/kibana/pull/14749. I will be merging all PRs into the home_phase_two as fast as possible to enable collaboration and development.

**Comment by nreese** [comment]
And please feel free to create PRs directly against https://github.com/elastic/kibana/pull/14749

**Comment by ycombinator** [comment]
Thanks @dedemorton for the feedback and thanks @nreese for incorporating it in! I'll make future PRs to elastic#14749 👍

**Pull request merged**